Alg_LZW/Trie/Trie.cs

```diff
@@ -0,0 +1,141 @@
+namespace Bor_Trie
+{
+    public class Trie
+    {
+        public class Node
+        {
+            public bool Terminal { get; set; }
+            public Dictionary<byte, Node> Children { get; set; }
+            public int Number { get; set; }
+            public int Code {get; set;}
+            public Node()
+            {
+                Terminal = false;
+                Children = new Dictionary<byte, Node>();
+                Number = 0;
+                Code = 0;
+            }
+        }
+
+        public Node root;
+        public int Size { get; set; }
+        public Trie()
+        {
+            Size = 0;
+            root = new Node();
+        }
+        public bool Add(List<byte> trybits, int code)
+        {
+            Node CheckNode = root;
+            for (int i = 0; i < trybits.Count; ++i)
+            {
+                if (CheckNode.Children.ContainsKey(trybits[i]))
+                {
+                    CheckNode = CheckNode.Children[trybits[i]];
+                }
+                else
+                {
+                    CheckNode.Children.Add(trybits[i], new Node());
+                    Size++;
+                    CheckNode = CheckNode.Children[trybits[i]];
+                }
+                CheckNode.Number++;
+            }
+            CheckNode.Code=code;
+            if (CheckNode.Terminal == false) 
```

Comment:
```suggestion
            if (!CheckNode.Terminal) 
```
правильно писать так